lib/Conversion/TorchToTcp/Elementwise.cpp

```diff
@@ -107,6 +127,126 @@ class ConvertAtenTanhOp : public OpConversionPattern<AtenTanhOp> {
+    IntegerAttr minIntAttr, maxIntAttr;
+
+    Value minValue = op.getMin();
+    Value maxValue = op.getMax();
+    if (checkNotNone(rewriter, op, minValue).failed() &&
+        checkNotNone(rewriter, op, maxValue).failed()) {
+      return rewriter.notifyMatchFailure(
+          op, "clamp op requires at least one of min or max");
+    }
+    if (checkNotNone(rewriter, op, minValue).succeeded()) {
+      double minFloat;
+      int64_t minInt;
+      if (matchPattern(minValue, m_TorchConstantFloat(&minFloat))) {
+        if (elementType.isa<mlir::FloatType>())
+          minFloatAttr = rewriter.getF32FloatAttr(minFloat);
+        else if (elementType.isa<mlir::IntegerType>())
+          minIntAttr =
+              rewriter.getI64IntegerAttr(static_cast<int64_t>(minFloat));
+      } else if (matchPattern(minValue, m_TorchConstantInt(&minInt))) {
+        if (elementType.isa<mlir::FloatType>())
+          minFloatAttr = rewriter.getF32FloatAttr(static_cast<float>(minInt));
+        else if (elementType.isa<mlir::IntegerType>())
+          minIntAttr = rewriter.getI64IntegerAttr(minInt);
+      } else {
+        return rewriter.notifyMatchFailure(
+            op, "expected to find a float or integer constant as min");
+      }
+    }
+    if (checkNotNone(rewriter, op, maxValue).succeeded()) {
+      double maxFloat;
+      int64_t maxInt;
+      if (matchPattern(maxValue, m_TorchConstantFloat(&maxFloat))) {
+        if (elementType.isa<mlir::FloatType>())
+          maxFloatAttr = rewriter.getF32FloatAttr(maxFloat);
+        else if (elementType.isa<mlir::IntegerType>())
+          maxIntAttr =
+              rewriter.getI64IntegerAttr(static_cast<int64_t>(maxFloat));
+      } else if (matchPattern(maxValue, m_TorchConstantInt(&maxInt))) {
+        if (elementType.isa<mlir::FloatType>())
+          maxFloatAttr = rewriter.getF32FloatAttr(static_cast<float>(maxInt));
+        else if (elementType.isa<mlir::IntegerType>())
+          maxIntAttr = rewriter.getI64IntegerAttr(maxInt);
+      } else {
+        return rewriter.notifyMatchFailure(
+            op, "expected to find a float or integer constant as max");
+      }
+    }
+
+    rewriter.replaceOpWithNewOp<tcp::ClampOp>(op, inputType, input,
+                                              minFloatAttr, maxFloatAttr,
+                                              minIntAttr, maxIntAttr);
+    return success();
+  }
+};
+
+template <typename AtenOpT>
+class ConvertAtenReluOp : public OpConversionPattern<AtenOpT> {
+public:
+  using OpConversionPattern<AtenOpT>::OpConversionPattern;
+  using OpAdaptor = typename AtenOpT::Adaptor;
+
+  LogicalResult
+  matchAndRewrite(AtenOpT op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Value input = adaptor.getSelf();
+    RankedTensorType inputType = input.getType().dyn_cast<RankedTensorType>();
+    auto elementType = inputType.getElementType();
+    if (!inputType)
```

Comment:
This should happen before `elementType` is defined

Reply:
Fixed